storage/src/commit_log/mod.rs

```diff
@@ -0,0 +1,430 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
```

Comment:
@mambisi @jurajselep 
Mambisi, imagine, that you want to introduce commit log to someone, who have never heard about it,
so please, add here some simple documentation:
- what is it used for
- how it is working
- what and how is stored
- key points
- caveats, specialities like that compression


maybe the commit_log is also not a good name,
we will see, when you update documentation here maybe.

it does not have to be article-like style, just few simple bullets is ok,
or something like here:
https://github.com/simplestaking/tezedge/blob/master/storage/src/merkle_storage.rs#L4-L47